Align UA Stylesheet for `merror` in mathml.css

Ahmad-S792 opened the pull request:
#### 76ad36c10382116e570ea836ac72bd0d37fcda54
<pre>
Align UA Stylesheet for `merror` in mathml.css

<a href="https://bugs.webkit.org/show_bug.cgi?id=262670">https://bugs.webkit.org/show_bug.cgi?id=262670</a>

Reviewed by Tim Nguyen and Frédéric Wang.

This patch aligns WebKit with Web-Specification [1]:

[1] <a href="https://w3c.github.io/mathml-core/#error-message-merror">https://w3c.github.io/mathml-core/#error-message-merror</a>

WebKit was using &apos;outline&apos; rather than &apos;border&apos; for `merror`, which is now updated in UA Stylesheet.

* Source/WebCore/css/mathml.css: As above
* LayoutTests/imported/w3c/web-platform-tests/mathml/presentation-markup/mrow/merror-001-expected.txt: Added Expectation File
* LayoutTests/platform/glib/imported/w3c/web-platform-tests/mathml/presentation-markup/mrow/merror-001-expected.txt: Rebaselined
* LayoutTests/platform/ios-wk2/imported/w3c/web-platform-tests/mathml/presentation-markup/mrow/merror-001-expected.txt: Removed
* LayoutTests/platform/mac-wk1/imported/w3c/web-platform-tests/mathml/presentation-markup/mrow/merror-001-expected.txt: Ditto
* LayoutTests/platform/mac-wk2/imported/w3c/web-platform-tests/mathml/presentation-markup/mrow/merror-001-expected.txt: Ditto
* LayoutTests/imported/w3c/web-platform-tests/mathml/relations/css-styling/padding-border-margin/border-002-expected.txt: Rebaselined
* LayoutTests/platform/gtk/imported/w3c/web-platform-tests/mathml/relations/css-styling/padding-border-margin/border-002-expected.txt: Rebaselined
* LayoutTests/platform/mac-monterey/imported/w3c/web-platform-tests/mathml/relations/css-styling/padding-border-margin/border-002-expected.txt: Rebaselined
* LayoutTests/platform/wpe/imported/w3c/web-platform-tests/mathml/relations/css-styling/padding-border-margin/border-002-expected.txt: Rebaselined
* LayoutTests/imported/w3c/web-platform-tests/mathml/presentation-markup/operators/embellished-operator-001-expected.txt: Rebaselined
* LayoutTests/imported/w3c/web-platform-tests/mathml/relations/css-styling/padding-border-margin/margin-003-expected.txt: Rebaselined
* LayoutTests/platform/glib/imported/w3c/web-platform-tests/mathml/presentation-markup/operators/embellished-operator-001-expected.txt: Rebaselined
* LayoutTests/platform/gtk/imported/w3c/web-platform-tests/mathml/presentation-markup/operators/embellished-operator-001-expected.txt: Removed
* LayoutTests/platform/ios-wk2/imported/w3c/web-platform-tests/mathml/presentation-markup/operators/embellished-operator-001-expected.txt: Ditto
* LayoutTests/platform/mac-wk1/imported/w3c/web-platform-tests/mathml/presentation-markup/operators/embellished-operator-001-expected.txt: Ditto
* LayoutTests/platform/mac-wk2/imported/w3c/web-platform-tests/mathml/presentation-markup/operators/embellished-operator-001-expected.txt: Ditto
* LayoutTests/platform/glib/imported/w3c/web-platform-tests/mathml/relations/css-styling/ignored-properties-001-expected.txt: Add Expectation (on Glib for WPE and GTK ports)
* LayoutTests/platform/glib/imported/w3c/web-platform-tests/mathml/relations/css-styling/padding-border-margin/margin-003-expected.txt: Ditto
* LayoutTests/platform/glib/TestExpectations: Updated
* LayoutTests/platform/gtk/imported/w3c/web-platform-tests/mathml/relations/css-styling/ignored-properties-001-expected.txt: Deleted in favor of &apos;GLIB&apos;
* LayoutTests/platform/gtk/imported/w3c/web-platform-tests/mathml/relations/css-styling/padding-border-margin/margin-003-expected.txt: Ditto
* LayoutTests/platform/gtk/TestExpectations: Updated
* LayoutTests/platform/ios/imported/w3c/web-platform-tests/mathml/relations/css-styling/ignored-properties-001-expected.txt: Rebaselined
* LayoutTests/platform/mac-monterey/imported/w3c/web-platform-tests/mathml/relations/css-styling/padding-border-margin/margin-003-expected.txt: Deleted &apos;Platform&apos; specific expectation
* LayoutTests/platform/mac/TestExpectations: Updated
* LayoutTests/imported/w3c/web-platform-tests/mathml/relations/css-styling/ignored-properties-001-expected.txt: Rebaselined
* LayoutTests/platform/mac/imported/w3c/web-platform-tests/mathml/relations/css-styling/padding-border-margin/margin-003-expected.txt: Add Platform Specific Expectation
* LayoutTests/platform/mac-ventura/imported/w3c/web-platform-tests/mathml/relations/css-styling/ignored-properties-001-expected.txt: Rebaselined
* LayoutTests/platform/mac-ventura/imported/w3c/web-platform-tests/mathml/relations/css-styling/padding-border-margin/margin-003-expected.txt: Added &apos;macOS Ventura&apos; specific expectation

Canonical link: <a href="https://commits.webkit.org/268941@main">https://commits.webkit.org/268941@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/9da618ee274ddf50f28105d55deb746275cf4523

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/21129 "Passed style check") | [  ~~🛠 ios~~](https://ews-build.webkit.org/#/builders/26/builds/21494 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 mac~~](https://ews-build.webkit.org/#/builders/14/builds/22192 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 wpe~~](https://ews-build.webkit.org/#/builders/5/builds/23006 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 wincairo~~](https://ews-build.webkit.org/#/builders/32/builds/19646 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/21369 "Passed tests") | [  ~~🛠 ios-sim~~](https://ews-build.webkit.org/#/builders/23/builds/24758 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 mac-AS-debug~~](https://ews-build.webkit.org/#/builders/16/builds/21690 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/5/builds/23006 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/21353 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/23/builds/24758 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/14/builds/22192 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 gtk~~](https://ews-build.webkit.org/#/builders/2/builds/23860 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/23/builds/24758 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/14/builds/22192 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/2/builds/23860 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/23/builds/24758 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/14/builds/22192 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/2/builds/23860 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🛠 tv~~](https://ews-build.webkit.org/#/builders/7/builds/19893 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/16/builds/21690 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [  ~~🛠 tv-sim~~](https://ews-build.webkit.org/#/builders/30/builds/19174 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk2-stress~~](https://ews-build.webkit.org/#/builders/14/builds/22192 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/4/builds/23461 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/2613 "Built successfully and passed tests") | [  ~~🛠 watch-sim~~](https://ews-build.webkit.org/#/builders/31/builds/19752 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
<!--EWS-Status-Bubble-End-->

nt1m commented on Source/WebCore/css/mathml.css:
```diff
@@ -175,8 +175,9 @@ mphantom {
 }
 
 /* This is a special style for erroneous markup. */
+/* Web Specification: https://w3c.github.io/mathml-core/#error-message-merror */
```
```suggestion
/* https://w3c.github.io/mathml-core/#error-message-merror */
```

Ahmad-S792 replied:
Done.

fred-wang commented on Source/WebCore/css/mathml.css:
```diff
@@ -175,8 +175,9 @@ mphantom {
-    outline: solid thin red;
+    border: 1px solid red;
     font-weight: bold;
     font-family: sans-serif;
```
Is it possible to remove `font-weight` and `font-family` too?

I guess this can be tested in `merror-001.html` by comparing against the `mrow`'s style?

Ahmad-S792 replied:
I did and now pushed another EWS run and there might be more failures but I will fix them in later push. :-)

fred-wang reviewed (approved):
Thanks.

Ahmad-S792 committed:
Align UA Stylesheet for `merror` in mathml.css

https://bugs.webkit.org/show_bug.cgi?id=262670

Reviewed by Tim Nguyen and Frédéric Wang.

This patch aligns WebKit with Web-Specification [1]:

[1] https://w3c.github.io/mathml-core/#error-message-merror

WebKit was using 'outline' rather than 'border' for `merror`, which is now updated in UA Stylesheet.

* Source/WebCore/css/mathml.css: As above
* LayoutTests/imported/w3c/web-platform-tests/mathml/presentation-markup/mrow/merror-001-expected.txt: Added Expectation File
* LayoutTests/platform/glib/imported/w3c/web-platform-tests/mathml/presentation-markup/mrow/merror-001-expected.txt: Rebaselined
* LayoutTests/platform/ios-wk2/imported/w3c/web-platform-tests/mathml/presentation-markup/mrow/merror-001-expected.txt: Removed
* LayoutTests/platform/mac-wk1/imported/w3c/web-platform-tests/mathml/presentation-markup/mrow/merror-001-expected.txt: Ditto
* LayoutTests/platform/mac-wk2/imported/w3c/web-platform-tests/mathml/presentation-markup/mrow/merror-001-expected.txt: Ditto
* LayoutTests/imported/w3c/web-platform-tests/mathml/relations/css-styling/padding-border-margin/border-002-expected.txt: Rebaselined
* LayoutTests/platform/gtk/imported/w3c/web-platform-tests/mathml/relations/css-styling/padding-border-margin/border-002-expected.txt: Rebaselined
* LayoutTests/platform/mac-monterey/imported/w3c/web-platform-tests/mathml/relations/css-styling/padding-border-margin/border-002-expected.txt: Rebaselined
* LayoutTests/platform/wpe/imported/w3c/web-platform-tests/mathml/relations/css-styling/padding-border-margin/border-002-expected.txt: Rebaselined
* LayoutTests/imported/w3c/web-platform-tests/mathml/presentation-markup/operators/embellished-operator-001-expected.txt: Rebaselined
* LayoutTests/imported/w3c/web-platform-tests/mathml/relations/css-styling/padding-border-margin/margin-003-expected.txt: Rebaselined
* LayoutTests/platform/glib/imported/w3c/web-platform-tests/mathml/presentation-markup/operators/embellished-operator-001-expected.txt: Rebaselined
* LayoutTests/platform/gtk/imported/w3c/web-platform-tests/mathml/presentation-markup/operators/embellished-operator-001-expected.txt: Removed
* LayoutTests/platform/ios-wk2/imported/w3c/web-platform-tests/mathml/presentation-markup/operators/embellished-operator-001-expected.txt: Ditto
* LayoutTests/platform/mac-wk1/imported/w3c/web-platform-tests/mathml/presentation-markup/operators/embellished-operator-001-expected.txt: Ditto
* LayoutTests/platform/mac-wk2/imported/w3c/web-platform-tests/mathml/presentation-markup/operators/embellished-operator-001-expected.txt: Ditto
* LayoutTests/platform/glib/imported/w3c/web-platform-tests/mathml/relations/css-styling/ignored-properties-001-expected.txt: Add Expectation (on Glib for WPE and GTK ports)
* LayoutTests/platform/glib/imported/w3c/web-platform-tests/mathml/relations/css-styling/padding-border-margin/margin-003-expected.txt: Ditto
* LayoutTests/platform/glib/TestExpectations: Updated
* LayoutTests/platform/gtk/imported/w3c/web-platform-tests/mathml/relations/css-styling/ignored-properties-001-expected.txt: Deleted in favor of 'GLIB'
* LayoutTests/platform/gtk/imported/w3c/web-platform-tests/mathml/relations/css-styling/padding-border-margin/margin-003-expected.txt: Ditto
* LayoutTests/platform/gtk/TestExpectations: Updated
* LayoutTests/platform/ios/imported/w3c/web-platform-tests/mathml/relations/css-styling/ignored-properties-001-expected.txt: Rebaselined
* LayoutTests/platform/mac-monterey/imported/w3c/web-platform-tests/mathml/relations/css-styling/padding-border-margin/margin-003-expected.txt: Deleted 'Platform' specific expectation
* LayoutTests/platform/mac/TestExpectations: Updated
* LayoutTests/imported/w3c/web-platform-tests/mathml/relations/css-styling/ignored-properties-001-expected.txt: Rebaselined
* LayoutTests/platform/mac/imported/w3c/web-platform-tests/mathml/relations/css-styling/padding-border-margin/margin-003-expected.txt: Add Platform Specific Expectation
* LayoutTests/platform/mac-ventura/imported/w3c/web-platform-tests/mathml/relations/css-styling/ignored-properties-001-expected.txt: Rebaselined
* LayoutTests/platform/mac-ventura/imported/w3c/web-platform-tests/mathml/relations/css-styling/padding-border-margin/margin-003-expected.txt: Added 'macOS Ventura' specific expectation

Canonical link: https://commits.webkit.org/268941@main
